### DISCUSSION TITLE
RFC: Which CMS could we use across a variety of projects?

### DISCUSSION THREAD

**Discussion by wpmonty** [BanklessDAO/bankless-dao-website#42] [Ideas]
Requirements for new projects are brewing.

We need to plan two steps ahead and consider which CMS (content management system) could fit the needs of a variety of projects - even those we haven't considered yet.

_There is an argument for using a different CMS for specific projects, but ideally we would try to herd content operations into a single CMS._

**Potential Projects**

- NFT museum (ancillary content)
- Blog posts
- Website announcements
- Landing page creation
- Partnership listings

**Considerations**

- i18n / translations
- high client-side availability or just build time
- onboarding time & seat management
- price
- extensibility
- API / GraphQL

**Some Options**

- headless WordPress (admin interface with API only)
- Contentful
- custom application
- none / manual code changes

*Please comment below with your experience and opinions on which CMS we could use.*

**Comment by lvnacy** [comment]
Ghost is an option as well for a CMS. Considering most, if not all, Bankless members are involved on the Discord server, it may make sense to handle authentication with a CMS that incorporates OAuth2, taking username/password security risks out of the equation.

**Reply by wpmonty** [reply]
very much agree about the OAuth. IIRC ghost is a WordPress fork, so the same headless/self-hosted idea there?

**Reply by eniko1556** [reply]
How does i18n works with ghost?

**Comment by Oktalize** [comment]
All great options so far, I definitely think a headless CMS is a clear winner here so I am glad to see that. I have also used Sanity a bit before, which may not be as powerful but could be a cheaper alternative to Contentful. 

Either way it is definitely worth considering getting the blogging side going fast with a different CMS for now just to get going. I know merging it later sucks but it also makes our hungry writers happy while the rest is built out. 

Worth a thought, I will keep thinking about this and see if I come up with any other good ideas.

**Comment by grendelorr** [comment]
A headless WordPress or similar could / would be the best. 
The easier and faster, the better to start.

**Comment by pnpancholi** [comment]
My vote goes to Contentful.

Reasons:

 1.   Language Agnostic
 2.   GraphQL available - would be easier to optimize when data growth.
 3.   Microservices architecture (available for free plan) - allows us to be scalable, some platforms limit perf on free accounts which is not the case here.

I am also not a big fan of audience limits like "100 Listeners" or  "X amount of CDN request", Which some other platforms have.

**Reply by eniko1556** [reply]
The free tier is only for 5 seats. I think that's too big of a constraint. @wpmonty do we have a budget for tools and services?

**Reply by wpmonty** [reply]
We do not. There are ongoing talks about funding meatspace expenses with some appetite from contributors for taking on the expenses on the individual scale and getting paid later. Bankless LLC also has a small appetite for handling some of this burden (for example, our public website hosting).

At high seat count, Contentful gets extremely expensive, but if money were no object I would put it as my primary choice.

**Reply by pnpancholi** [reply]
Yeah, pricing is really tough there... actually most expensive. I looked at Prismic, Ghost, and Strapi. Strapi self-hosted looks the most affordable while still providing us everything we need. Do we have an estimate of how many seats/users we will have at least in the beginning? 

https://strapi.io/pricing-self-hosted.

Let me know your thoughts.

**Comment by pdarden** [comment]
I've used Contentful and [NetlifyCMS](https://www.netlifycms.org/).  Both are great options, but in my experience NetlifyCMS is more non-dev user friendly.

**Comment by jonathanprozzi** [comment]
[Sanity](https://www.sanity.io/) is another option to consider. I've not used it on a larger project but it integrates very well with Next.js
In my experience it's a bit strange to set up and get accustomed to, but extremely powerful once configured. It has it's own separate query language similar to GraphQL. It may be a bit too much but it has very advanced content modeling capabilities.

I've had great experiences with using headless WordPress too. 

Another (very new!) option is using Notion. There have been some projects in the past using Notion as a CMS (mostly relying on SSG in Next) but Notion just launched a public beta of their API. I'm exploring this for some personal projects. I've used the Next + Notion combo before and it's great but really shines when folks are already using Notion as their content hub.

**Reply by wpmonty** [reply]
Love the idea of Notion here, especially as we ingrain it in the rest of the guilds.

**Reply by pnpancholi** [reply]
Yeah, that might actually work out great. Will look into it

**Reply by jonathanprozzi** [reply]
I have some resources I can share. Have a site I did prior to the api release but it uses Next ssg and the build time is fairly long. Not at my laptop until later but will share when I'm back

**Reply by jonathanprozzi** [reply]
For the 2 Nextjs/Notion sites that I have deployed, I modified this Nextjs starter built by Travis Fischer: https://github.com/transitive-bullshit/nextjs-notion-starter-kit - This starter was created and released prior to Notion's public API that just went live in beta 2 weeks ago.

It's great since it pulls content from Notion but it uses Next's SSG. Being able to publish content from Notion was great, but the drawback was that it required a build step to push/display new or updated content (like with all static site generators -- I definitely have a love/hate relationship with them!) There's some newer approaches we could consider too such as Next's _Incremental Static Regeneration_ (https://nextjs.org/docs/basic-features/data-fetching#incremental-static-regeneration). I haven't had a chance to use this yet but I've seen demos and it's awesome.

The other option is to use SSR and fetch content from Notion with getServerSideProps if we wanted to fully use SSR and pull in content as folks make changes in Notion (without any build step).

**Reply by wpmonty** [reply]
ISR is the best of both worlds. Unfortunately, Netlify doesn't offer it out of the box. Closest would be their On-demand Builder (beta) https://docs.netlify.com/configure-builds/on-demand-builders/

Meanwhile, SSG build-times shouldn't be too bad. I'm primarily concerned with the effectiveness of Notion as a CMS from an end-user standpoint - and if its sustainable from a cost/user perspective if we scale.

**Reply by pnpancholi** [reply]
notions looks good so far, we can probably give it a try by doing something experimental and simple on Bankless DAO website.

**Comment by thetechnocrat-dev** [comment]
I've used Contentstack (https://www.contentstack.com/) as a headless CMS in production with React and it worked well. Both for the development team and our marketing team that manages the content.

**Reply by wpmonty** [reply]
Looks like they support Locales which is good, but $995/mo base price is steep.

**Comment by pnpancholi** [comment]
So far Strapi looks great to be price-wise if Notion doesn't work out for us. I would love Notion to work for us as we are already paying for it and the non-dev members would be more comfortable with it.

**Comment by unbanksytv** [comment]
I would love if we could (re)consider using Ghost as headless CMS in a JAMstack context

**Reply by pnpancholi** [reply]
Do you think that the limited views might be an issue? And number of seats will also be an issue.

**Reply by wpmonty** [reply]
Self hosting Ghost would be the likely solution there. The one caveat we have with Ghost is that it's primarily for writers (ex Blog), not so much for custom content types (ex Onboard). Still, if we could even stick to two CMS's - one for each purpose - I think that would be OK.
